maintenance/image-options/imageset.md

```diff
@@ -0,0 +1,227 @@
+        `example.com/registry/imagepath/node` you will still use `calico/node` in the image field of the ImageSet.
+        >**Example**: For image `quay.io/tigera/operator@sha256:d111db2f94546415a30eff868cb946d47e183faa804bd2e9a758fd9a8a4eaff1` copy everything after `@` and add it as the digest for the `tigera/operator` image.
+        {: .alert .alert-info}
+
+%>
+<label:Script>
+<%
+
+Copy the following script into a file, make it executable, and run the script. The script creates an `imageset.yaml` file in the directory it was run.
+>**Note**: This script will only work if using the default registries and image paths.
+{: .alert .alert-info}
+
+```
+#!/bin/bash -e
+
+images=(calico/cni calico/kube-controllers calico/node calico/typha calico/pod2daemon-flexvol tigera/key-cert-provisioner tigera/operator)
+
+OPERATOR_IMAGE={{ operator.registry }}/{{ operator.image }}:{{ operator.version }}
+echo "Pulling $OPERATOR_IMAGE"
+echo
+docker pull $OPERATOR_IMAGE -q >/dev/null
+versions=$(docker run $OPERATOR_IMAGE --version)
+ver=$(echo -e "$versions" | grep 'Calico:')
+
+imagelist=($(docker run $OPERATOR_IMAGE --print-images=list))
+
+cat > ./imageset.yaml <<EOF
+apiVersion: operator.tigera.io/v1
+kind: ImageSet
+metadata:
+  name: calico-$(echo $ver | sed -e 's|^.*: *||')
+spec:
+  images:
+EOF
+
+for x in "${imagelist[@]}"; do
+  for y in ${images[*]}; do
+    if [[ $x =~ $y: ]]; then
+      echo "Pulling $x"
+      docker pull $x -q > /dev/null
+      {% raw %}digests=$(docker inspect $x -f '{{range .RepoDigests}}{{printf "%s\n" .}}{{end}}'){% endraw %}
+      name=$(echo $x | sed -e 's|docker.io/||' -e 's|:.*$||')
+      imgdigest=$(echo -e $digests | grep $name)
+      digest=$(echo -e $imgdigest | sed -e 's|^.*@||')
+      echo "Adding digest for $x"
+      echo 
+      echo "  - image: \"$(echo $x | sed -e 's|^.*/\([^/]*/[^/]*\):.*$|\1|')\"" >> ./imageset.yaml
+      echo "    digest: \"$digest\"" >> ./imageset.yaml
+    fi
+  done
+done
+```
+%>
+{% endtabs %}
+
+Apply the created `imageset.yaml` to your cluster.
+
+#### Verification steps
+
+1. Check tigerastatus for components that are Degraded with `kubectl get tigerastatus`.
+   - If any components show Degraded, [investigate further](#how-can-i-tell-if-there-is-a-problem-with-my-imageset).
+2. When tigerastatus for all components show Available True, the ImageSet has been applied.
+   ```
+   NAME     AVAILABLE   PROGRESSING   DEGRADED   SINCE
+   calico   True        False         False      54s
+   ```
+3. Verify that the correct ImageSet is being used. In Installation status, check that the `imageset` field is set to the ImageSet you created.
+   Check the field by running the following command:
+   ```
+   kubectl get installation default -o yaml | grep imageSet
+   ```
+   You should see output similar to:
+   ```
+       imageSet: calico-{{site.data.versions.first.title}}
+   ```
+
```

Comment:
I would add a heading here to match list...found myself wondering what these sections were.

#### Other tasks

#### Troubleshooting